test/Franchiser.t.sol

```diff
@@ -203,6 +211,55 @@ contract FranchiserTest is Test, IFranchiserErrors, IFranchiserEvents {
         assertEq(votingToken.balanceOf(address(returnedFranchiser)), 50);
     }
 
+    function testFuzz_SubDelegateBalancesUpdated(
+        address _delegator,
+        address _delegatee,
+        address _subDelegatee,
+        uint256 _amount
+    ) public {
+        vm.assume(_validActorAddress(_delegator));
+        vm.assume(_delegatee != address(0));
+        vm.assume(_subDelegatee != address(0));
+        vm.assume(_delegatee != _subDelegatee);
+        _amount = _boundAmount(_amount);
+        franchiser.initialize(_delegator, _delegatee, 1);
+        votingToken.mint(address(franchiser), _amount);
+
+        Franchiser _expectedSubFranchiser = franchiser.getFranchiser(_subDelegatee);
+        uint256 _subFranchiserBalanceBefore = votingToken.balanceOf(address(_expectedSubFranchiser));
+        uint256 _franchiserBalanceBefore = votingToken.balanceOf(address(franchiser));
+        uint256 _delegateeVotingPowerBefore = votingToken.getVotes(_delegatee);
+        uint256 _subDelegateeVotingPowerBefore = votingToken.getVotes(_subDelegatee);
+        vm.prank(_delegatee);
+        Franchiser subFranchiser = franchiser.subDelegate(_subDelegatee, _amount);
+
+        assertEq(votingToken.balanceOf(address(subFranchiser)), _subFranchiserBalanceBefore + _amount);
+        assertEq(votingToken.balanceOf(address(franchiser)), _franchiserBalanceBefore - _amount);
+        assertEq(votingToken.getVotes(_delegatee), _delegateeVotingPowerBefore - _amount);
+        assertEq(votingToken.getVotes(_subDelegatee), _subDelegateeVotingPowerBefore + _amount);
+    }
+
+    function testFuzz_RevertIf_SubDelegateCalledWithFranchiserBalanceTooLow(
+        address _delegator,
+        address _delegatee,
+        address _subDelegatee,
+        uint256 _amount,
+        uint256 _delta
+    ) public {
+        vm.assume(_validActorAddress(_delegator));
+        vm.assume(_delegatee != address(0));
+        vm.assume(_subDelegatee != address(0));
+        vm.assume((_amount >= _delta) && (_amount <= 100_000_000e18));
```

Comment:
Is this necessary with the bound statements below

Reply:
Good catch.. not needed.. thanks! 77bf13e772becab7b48d4a0f10297b36058d5f63